meetings/working-groups/collection-literals/LDM-questions-2023-08-15.md

```diff
@@ -0,0 +1,223 @@
+        {
+            // 's' has 'global scope' but does not escape.  Will allocate.  If they want stack-allocation, add `scoped`
+            Span<int> s = [x, y, z];
+            foreach (var v in s)
+                Console.WriteLine(v);
+        }
+        ```
+
+        Ideally compiler tells the user to make these variables `scoped` when it can be.
+
+        Pros: User gets the scoping they explicitly request.  When scope can be narrower, they are informed.
+
+        Cons: Analysis is similarly complex due to requisite reverse flow analysis.  Also, WG believes the default position of users is to want a span + collection-expression to have local scope and be stack allocated.  However, to get that, users will have to write:
+
+        ```c#
+        void M()
+        {
+            // 's' has 'local scope'
+            scoped Span<int> s = [x, y, z];
+            foreach (var v in s)
+                Console.WriteLine(v);
+        }
+        ```
+
+        This feels punishing to the common/default case that users will want.  Importantly, it means that if a user wants to switch off from `stackalloc` to collection-expressions they get less pleasant code. 
+
+    - Option B: Ref-struct local always has local scope.
+
+        ```c#
+        void M()
+        {
+            // 's' has 'local scope'.  Will get stack allocated.
+            Span<int> s = [x, y, z];
+            foreach (var v in s)
+                Console.WriteLine(v);
+        }
+        ```
+
+        Pros: A very simple and easy rule to explain to users.
+
+        Default, simple, syntax gives best performance.
+
+        No need for any sort of complex flow analysis.  Existing analysis "just works" and lets the user know if there are any problems.
+
+        When code needs to allocate on the heap it is clear in the code that this is happening.
+
+        Cons: if the user *does* want the span to have global scope, then they have to be explicit about that:
+
+        ```c#
+        Span<int> M()
+        {
+            // Either:
+            Span<int> s1 = (int[])[x, y, z];
+            Span<int> s1 = new[] { x, y, z };
+            
+            return x ? s1 : s2;
+        }
+        ```
+
+        However, WG believe this is a good thing.  This will be the rare case, and having *it* be slightly more verbose is unlikely to be a burden.  In other words, we want the common, fast, case to sing with collection-expressions, while the less common, slower, case is ok to pay a small price.
+
+Working group suggestion:
+
+Option 2B. Collection-expressions of ref-struct type have local-scope.
+
+### Overload resolution priority
+
+We know we want a `Span<T>` overload to be preferred over a `T[]` overload or an interface overload ( `IEnumerable<T>/IReadOnlyCollection<T>/IReadOnlyList<T>/ICollection<T>/IList<T>`).
+
+However, we have options on how to specify things to get the above outcome.
+
+1. Option A: Span/ReadOnlySpan is better than arrays and those specific interfaces.
```

Comment:
I think we need to be clear that the *iteration types* must be compatible. Perhaps:

```suggestion
1. Option A: Span/ReadOnlySpan is preferred over an array or specific interface type when the iteration type of the span is implicitly convertible to the iteration type of the array or interface.
```